app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -5,7 +5,8 @@ module Foreman::Controller::RegistrationCommands
 
   def command
     args_query = "?#{registration_args.to_query}"
-    "curl -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers} | bash"
+    "tmp=$(mktemp) && curl -o $tmp -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers} && bash $tmp; CODE=$?; rm -f $tmp; (exit $CODE)
```

Comment:
@ekohl Need some bash expertise here: 
TL;DR: we want to fail the script if the `curl` command fails. The original solution `curl... | bash` doesn't work (it will always return status 0). 

Is this an acceptable solution, or is it a better one that we don't recognize?

Reply:
My two cents:
The solution in the PR is not wrong or bad IMO, it just adds a lot more complexity to the command, which I don't think is a good user experience.
I was able to achieve desired behavior with `set -o pipefail`

```
set -o pipefail
curl -sSk http://zz.redhat.com | bash;echo $?
=>
curl: (6) Could not resolve host: zz.redhat.com
6
```

Technically speaking, aren't we trying to fix behavior that is expected and correct? 
Maybe the fix is to configure user's CI pipeline with `set -o pipefail`

Reply:
I'd say using `pipefail` is correct (it's also what Ansible suggests in their linter: https://ansible.readthedocs.io/projects/lint/rules/risky-shell-pipe/) and that's what we should incorporate in the generated command?

So make the PR here generate something like
```
"set -o pipefail && curl -sS #{insecure} '#{registration_url(@smart_proxy)}#{args_query if args_query != '?'}' #{command_headers} | bash"
```

Reply:
> that's what we should incorporate in the generated command?

IMHO we could just add it to the documentation.

Reply:
Why docs instead of here?

Reply:
to keep the command short as possible.
I know that with current length of the auth token it's not a big addition, but still ...

Reply:
I had a look at it. IMO `set -o pipefail` is definitely the way we should go. Personally I would also tend to extend the command as proposed by @evgeni. I have the feeling that adding it to the docs is a little bit unwieldy. Yes, it is only one extra bullet point in a list of 25 you have to follow when registering a host, but it's somehow interrupting the workflow between copying the generated curl command and pasting it on the host.

Reply:
I am also for automatic actions - the less we expect from the user, the less are chances for command failure.